slick/src/main/scala/slick/ast/Node.scala

```diff
@@ -385,6 +385,16 @@ final case class ForUpdate(generator: TermSymbol, from: Node) extends ComplexFil
   protected[this] def rebuildWithSymbols(gen: ConstArray[TermSymbol]) = copy(generator = gen(0))
 }
 
+/** A .headOption call. */
+final case class HeadOption(from: Node) extends SimplyTypedNode with UnaryNode {
```

Comment:
`FirstOption` is a `ClientSideOp` that does the final conversion from the `ResultSetMapping` when `result` is called on the query. This is useful for wrapping a call similar to `query.headOption.result`.

`HeadOption` is an internal AST node that represents a `query.headOption` call.